### ISSUE TITLE
[Feature Request]: Support Baka MPlayer, Celluloid, and Haruna external media players. (various libmpv-based media players)

### ISSUE THREAD

**Issue by Cyanide-1** [FreeTubeApp/FreeTube#3990]
### Guidelines

- [X] I have searched the issue tracker for [open](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aopen+is%3Aissue) and [closed](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aissue+is%3Aclosed) issues that are similar to the feature request I want to file, without success.
- [X] I have searched the [documentation](https://docs.freetubeapp.io/) for information that matches the description of the feature request I want to file, without success.
- [X] This issue contains only one feature request.

### Problem Description

I would like Baka MPlayer, Celluloid, and Haruna external media players to be supported as media player options.

### Proposed Solution

Support Baka MPlayer, Celluloid, and Haruna media players.

### Alternatives Considered

I suppose the alternative is simply not supporting it.

### Issue Labels

improvement to existing feature, new optional setting, support for external software

### Additional Information

All 3 of these media players natively support extracting youtube videos and playing them with a direct link, and they all do this in the same exact way, as long as yt-dlp or similar is installed, they will automatically call yt-dlp to extract the video and then play it in the media player, the command can be ran in the terminal like "(baka-mplayer or celluloid or haruna) (youtube video link)", similar to how mpv itself does it.

Implementing this support i would assume would be relatively simple, since these programs natively deal with the youtube video stuff on their own, and since these are based on libmpv i assume that it may even just be as simple as copy-pasting the code from that already-supported media player, although i am not a programmer so dont take my word on that 100%, there may or may not be differences that have to be accounted for when implementing this support.

** Links to the projects in question: **

Baka MPlayer: https://u8sand.github.io/Baka-MPlayer/

Celluloid: https://celluloid-player.github.io/

Haruna: https://invent.kde.org/multimedia/haruna

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
@Cyanide-1 u have checked `This issue contains only one feature request` but you are requesting multiple external players to be implemented. I'll allow it this time but keep in mind for the next time you're creating issues that implementing one this = one issue

**Comment by Cyanide-1** [comment]
Sorry about the multiple features in one request, i thought that since all the players were gonna be supported the same/similar way and since they were all libmpv-based that it would be fine to put them all into 1 single feature request thing, i wont do it again, i will keep it in mind for next time

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
You have to look it from the angle of the dev (possibility a first time contributor) that has is looking for an issue to take on. They might take this one but get discouraged because they only get one of the three players to work and then they abandon the issue because it is requesting for all three players to work

**Comment by Cyanide-1** [comment]
@efb4f5ff-1298-471a-8973-3d47447115dc yeah you are right, should i split this up into 3 different issues then?

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Lets wait a bit until we have a response from @trostboot (contributor that add allot of support for external players in the past) before we chop this up

**Comment by trostboot** [comment]
I don't mind doing it and having these all in one issue, but I want to make a change to the list first, just haven't gotten around to it yet. In the meantime @Cyanide-1 , SMPlayer is a wrapper around mpv and is already supported.

**Comment by Cyanide-1** [comment]
thanks

**Comment by via80HD** [comment]
**I concur!** ![alt** text](https://keepmeme.com/files/en_posts/20200826/29d41a4cf7491b805ed90857bbc32471thumbs-up-like-with-a-face.jpg)

**Comment by trostboot** [comment]
@Cyanide-1 Please test the PR if it works for you, especially Baka. You can find a build [here](https://github.com/trostboot/FreeTube/actions/runs/6291285854).

**Comment by Cyanide-1** [comment]
> @Cyanide-1 Please test the PR if it works for you, especially Baka. You can find a build [here](https://github.com/trostboot/FreeTube/actions/runs/6291285854).

Sorry, i dont have a working computer at the moment, ill test it out as soon as possible though, hopefully i can test it out tomorrow, recently moved into a different room in the house so my stuff is still all over the place, PC isnt setup since no place to put it yet (gotta move the desk in), i'll test it out at some point though :)

**Comment by trostboot** [comment]
That's not unexpected as Baka hasn't seen any new development in years. Actual video playback is entirely down to the individual players, all FreeTube does is hand over the video/playlist URL and potentially parameters like playlist shuffle, repeat or playback speed if set in FreeTube and supported by the players commandline arguments.

Baka and Haruna only provide barebones options, while Celluloid supports setting the full range of mpv options. This should probably be displayed in the GUI in some form, but that's an issue for another day.

All that to say, thanks for testing.

**Comment by Cyanide-1** [comment]
> That's not unexpected as Baka hasn't seen any new development in years. Actual video playback is entirely down to the individual players, all FreeTube does is hand over the video/playlist URL and potentially parameters like playlist shuffle, repeat or playback speed if set in FreeTube and supported by the players commandline arguments.
> 
> Baka and Haruna only provide barebones options, while Celluloid supports setting the full range of mpv options. This should probably be displayed in the GUI in some form, but that's an issue for another day.
> 
> All that to say, thanks for testing.

no problem! thanks for adding the support!

**Comment by absidue** [comment]
As it sounds like Baka doesn't support YouTube videos and will likely not get any updates to support it, I think it is best if we don't add it to FreeTube, as it sounds like it won't be usuable anyway.
